src/modules/site-v2/base/views/data/data.py

```diff
@@ -57,3 +76,186 @@ def protocols():
     }
     return render_template('data/protocols.html', **params)
 
+
+#
+# Submit Trait
+#
+@data_bp.route('/submit-trait')
+@jwt_required()
+def submit_trait_start():
+  """ Submit Trait start page """
+  return render_template('data/submit-trait-start.html', **{
+    'title': 'Submit Trait',
+    'disable_parent_breadcrumb': True
+  })
+
+#
+# Submit Trait Form
+#
+@data_bp.route('/submit-trait/new-submission', methods=['GET', 'POST'])
+@jwt_required()
+def submit_trait_form():
+  """ Trait Submission Form """
+  form = TraitSubmissionForm()
+  user = get_current_user()
+
+  if hasattr(user, 'username') and not form.username.data:
+    form.username.data = user.username
+
+  # Handle form submission
```

Comment:
an architecture suggestion -- borrowing a page from the [MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller), see if we can decouple the `controller` role of this function in handling the HTTP request from the data manipulation that we need to apply `model`. In this case, consider creating another function that handles the creation of a trait where it has all the business logic to validate and create the trait artifacts (data store, file uploads, etc). One side benefit is that we could then call the trait function individually for automated tests while mocking the request payload without having to call this function directly.

Reply:
Pushed in the commit 2e21cc919bfb4210ba96e2b1e215fba0508dce51